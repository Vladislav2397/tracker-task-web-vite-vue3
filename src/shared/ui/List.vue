<template>
    <div class="list">
        <slot v-for="item in props.list" :item="item" />
    </div>
</template>

<script lang="ts" setup>
const props = defineProps<{ list: any[] }>()
</script>

<style scoped>
.list {
    display: flex;
    flex-direction: column;
    gap: 8px;
    width: 100%;
}
</style>
