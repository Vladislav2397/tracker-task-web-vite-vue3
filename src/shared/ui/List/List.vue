<template>
    <div class="list" v-auto-animate>
        <slot v-for="item in props.list" :key="props.keyExtractor(item)" :item="item" />
    </div>
</template>

<script lang="ts" setup>
const props = defineProps<{ list: any[], keyExtractor: (item: any) => string }>()
</script>

<style scoped>
.list {
    display: flex;
    flex-direction: column;
    gap: 8px;
    width: 100%;
}
</style>
