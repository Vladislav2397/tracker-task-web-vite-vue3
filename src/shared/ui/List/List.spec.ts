import { render } from '@testing-library/vue'
import List from './List.vue'

describe('shared/ui/List component', () => {
    it('Snapshot', () => {
        const { html } = render(List)

        expect(html()).toMatchSnapshot()
    })
})
