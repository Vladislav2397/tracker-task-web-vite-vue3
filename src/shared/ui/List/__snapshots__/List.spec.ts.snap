// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`shared/ui/List component > Snapshot 1`] = `"<div data-v-62220b6c=\\"\\" class=\\"list\\"></div>"`;
